### PR TITLE
Flexbox mixins & helper classes for direction, container, …

### PR DESCRIPTION
**Pull request by kball**
First pass on some mixins and helper classes for managing flex alignment etc.  Addresses https://github.com/zurb/foundation-sites/issues/9208

### CODE REVIEW

**Comment by brettsmason** [comment]
I'm still getting to grips with flexbox, but this looks brilliant from my first look over everything 👍

**Review by ncoden — APPROVED** [review]

File: docs/pages/flexbox.md [hunk 1]
```diff
@@ -163,6 +163,40 @@ To align an individual child, use the below classes. They use the same alignment
...
+
+To make something a flex container, simply apply
+- `.flex-container`
+
+And to change its flex direction from row to column you can use the helper classes:
+
+- `.flex-direct-row` (default)
+- `.flex-direct-row-reverse`
+- `.flex-direct-column`
+- `.flex-direct-column-reverse`
```
Comment (rafibomb) [inline docs/pages/flexbox.md]: Can we call it flex-dir-row? Dir is more shorthand direct and dont want to type out direction 😜

File: scss/grid/_flex-grid.scss [hunk 2]
```diff
@@ -200,6 +200,29 @@
...
+    @include flex;
+  }
+
+  .flex-child-auto {
+    flex: 1 1 auto;
+  }
+
+  .flex-child-grow {
+    flex: 1 0 auto;
+  }
```
Comment (rafibomb) [inline scss/grid/_flex-grid.scss]: Might need a media query for these. Like `.small-flex-child-auto` & `.medium-flex-child-grow`

File: docs/pages/flexbox.md (hunk 1 above)
Comment (kball) [inline docs/pages/flexbox.md]: sure thing

File: scss/grid/_flex-grid.scss (hunk 2 above)
Comment (kball) [inline scss/grid/_flex-grid.scss]: added

**Comment by kball** [comment]
Ok based on @ncoden's thumbsup and @rafibomb's review going to go ahead and merge this.

**Pull request merged**